### ISSUE TITLE
Test 'testEnteringReturnCharacterIntoViewWithAccessibilityLabel' is fails sometimes

### ISSUE THREAD

**Issue by brightredchilli** [kif-framework/KIF#560]
On my machine, the test above fails on iPhone 4s(7.1), but passed on iPad 2(7.1). The failure on the 4s is due to the fact that when the greeting label gets focus(on iPhone), we somehow get the cursor to the start of the label, causing the text to be entered to be ', worldHello', instead of the expected 'Hello, world'. I haven't begun to pin down what causes this behaviour to be different on the iPhone vs iPad but I suspect somehow the accessibilityFrames are different. This also does not seem to happen on Travis. 

Mysteriously, testEnteringTextIntoViewWithAccessibilityLabelExpectingResults seems to work even though it's almost the exact same test.

Does anyone have any  ideas why this may be happening and how to fix? One suggestion is to ensure that we wipe the text field clean before popping in any changes, after all, the test is testing entering return character, not that we append to the end successfully.

**Comment by brightredchilli** [comment]
This failure mentioned in #559

**Comment by phatmann** [comment]
I actually am seeing this fail on Travis occasionally. Can you recommend any way for me to reproduce this consistently so I can debug and fix it?

**Comment by brightredchilli** [comment]
Try running locally(iPhone 4s(7.1))

**Comment by phatmann** [comment]
I am indeed seeing the failure in the simulator running iPhone 4S under 7.1. I will look into it. If anyone else has an idea of what could be causing it, let me know.
